### PR TITLE
enhancement: Support optional `measure_name` in `wr.timestream.write()`

### PR DESCRIPTION
**Pull request by malachi-constant**
### Enhancement 
- Adds new _optional_ parameter `measure_name` to `wr.timestream.write()`.
- Column `measure_name` can be named to a string value that does not come from any measure column names. 
  - This is intended for use with multi-measure records , but is available as an override for single (scalar) records as well.
- Unit tests for both multi-measure and single record puts are included.

### Relates
- #1885 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: bab7bab855e0555f158f58b45b8df0633090b083
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b021e280-f7ad-4004-8523-bbfa73b4846a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 0083cd205edd360a23abb5a502da5360d2149713
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=18c4a914-4a80-4e76-9c91-9d1ab38dac7c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b30f497509b546a919f2aed1c41e802b789a6e43
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=368a408d-2dd4-413f-8e03-2a57ca8c65c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tests/test_timestream.py [hunk 1]
```diff
@@ -345,3 +345,67 @@ def test_timestamp_measure_column(timestream_database_and_table):
         """,
     )
     assert df["measure_t"].dtype == "datetime64[ns]"
+
+
+def test_measure_name_multi(timestream_database_and_table):
```
Comment (jaidisido) [inline tests/test_timestream.py:350]: The two tests are mostly similar so perhaps we can consolidate with a boolean parameter and then have an if condition:
```
data = {"time": [datetime.now()] * 3}
args = {
    "database": timestream_database_and_table,
    "table=timestream_database_and_table,
    "time_col="time",
}

if multi:  # multi being the boolean parameter
    data.update({   
        "dim0": ["foo", "boo", "bar"],
        "dim1": [1, None, 3],
        "measure_0": [1.1, 1.2, 1.3],
        "measure_1": [2.1, 2.2, 2.3],
    })
    args.update(...)
else:
    ...
```

**Review by jaidisido — COMMENTED** [review]
Looks good, minor refactor in tests

**Review by kukushking — APPROVED** [review]
Thanks!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 0d9632b21254072028a8aabd8327f6f3d2e430e2
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=64b97957-d167-4860-bef9-334253bc0dc0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 0d9632b21254072028a8aabd8327f6f3d2e430e2
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e77b6edd-fae9-4a9b-8e0f-866fd9d26345%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**